Samples/HelloWorld/NonSiloTests/TestGrainFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using Orleans;
+using Orleans.Core;
+using Orleans.Runtime;
+using Orleans.Storage;
+
+namespace NonSiloTests
+{
+    public static class TestGrainFactory
+    {
+        private static readonly GrainCreator GrainCreator;
+
+        private static readonly IStorageProvider StorageProvider;
+
+        static TestGrainFactory()
+        {
+            var grainRuntime = Mock.Of<IGrainRuntime>();
+
+            var mockStorageProvider = new Mock<IStorageProvider>();
+
+            StorageProvider = mockStorageProvider.Object;
+
+            GrainCreator = new GrainCreator(grainRuntime);
```

Comment:
oh, now I make the connection from your other PR... you are sill using the concrete GrainCreator here, and that's why it creates the GrainStateStorageBridge instead of being able to mock IStorage... I'll try to review the other PR in this light